src/meta/src/manager/sink_manager.rs

```diff
@@ -0,0 +1,578 @@
+    StopCoordinator {
+        finish_notifier: Sender<()>,
+        /// sink id to stop. When `None`, stop all sink coordinator
+        sink_id: Option<SinkId>,
+    },
+}
+
+#[derive(Clone)]
+pub struct SinkManager {
+    request_tx: UnboundedSender<SinkManagerRequest>,
+}
+
+impl SinkManager {
+    pub(crate) fn start_worker(
+        connector_client: Option<ConnectorClient>,
+    ) -> (Self, (JoinHandle<()>, Sender<()>)) {
+        let (request_tx, request_rx) = unbounded_channel();
+        let (shutdown_tx, shutdown_rx) = channel();
+        let worker = SinkManagerWorker {
+            request_rx,
+            shutdown_rx: Some(shutdown_rx),
+            running_coordinator_worker_join_handles: Default::default(),
+            running_coordinator_worker: Default::default(),
+            connector_client,
+        };
+        let join_handle = tokio::spawn(worker.execute());
+        (SinkManager { request_tx }, (join_handle, shutdown_tx))
+    }
+
+    pub(crate) async fn handle_new_request(
+        &self,
+        mut request_stream: SinkWriterInputStream,
+    ) -> Result<impl Stream<Item = Result<SinkCoordinatorToWriterMsg, Status>>, Status> {
+        let (param, vnode_bitmap) = match request_stream.try_next().await? {
+            Some(SinkWriterToCoordinatorMsg {
+                msg:
+                    Some(Msg::StartRequest(sink_writer_to_coordinator_msg::StartCoordinationRequest {
+                        param: Some(param),
+                        vnode_bitmap: Some(vnode_bitmap),
+                    })),
+            }) => (SinkParam::from_proto(param), Bitmap::from(&vnode_bitmap)),
+            msg => {
+                return Err(Status::invalid_argument(format!(
+                    "expected SinkWriterToCoordinatorMsg::StartRequest in the first request, get {:?}",
+                    msg
+                )));
+            }
+        };
+        let (response_tx, response_rx) = unbounded_channel();
+        self.request_tx
+            .send(SinkManagerRequest::NewSinkWriter(NewSinkWriterRequest {
+                request_stream,
+                response_tx,
+                param,
+                vnode_bitmap,
+            }))
+            .map_err(|_| {
+                Status::unavailable(
+                    "unable to send to sink manager worker. The worker may have stopped",
+                )
+            })?;
+
+        Ok(UnboundedReceiverStream::new(response_rx))
+    }
+
+    async fn stop_coordinator(&self, sink_id: Option<SinkId>) {
+        let (tx, rx) = channel();
+        let request_tx = &self.request_tx;
+        send_with_err_check!(
+            request_tx,
+            SinkManagerRequest::StopCoordinator {
+                finish_notifier: tx,
+                sink_id,
+            }
+        );
+        if rx.await.is_err() {
+            error!("fail to wait for resetting sink manager worker");
+        }
+    }
+
+    pub(crate) async fn reset(&self) {
```

Comment:
The `reset` will stop all the tokio tasks that drives the sink coordinator. It will notify the sink manager worker to do all the cleaning and notify back when the cleaning is finished. If we simply drop the sink manager and recreate a new one, we cannot make sure that we have finished cleaning.